# Project 10 -- Arnav Surve

**TA Help:** N/A
    
**Collaboration:** N/A

## Question 1

In [1]:
options(jupyter.rich_display = F)

In [2]:
options(repr.matrix.max.cols=25, repr.matrix.max.rows=200)

In [3]:
library(data.table)

In [4]:
users <- fread("/anvil/projects/tdm/data/okcupid/filtered/users.csv")

In [5]:
questions <- fread("/anvil/projects/tdm/data/okcupid/filtered/questions.csv")

In [8]:
dim(users)

[1] 68371  2284

In [6]:
head(users)

  q41                  q42 q43 q45 q46         q48   q49      q50 q55 q56 ...
1 NA                   NA  NA  NA  NA          NA    Carefree NA  No  No  ...
2 NA                   NA  NA  NA  NA          NA    NA       NA  NA  NA  ...
3 Extremely important  NA  NA  NA  Good        NA    Carefree NA  No  NA  ...
4 NA                   NA  NA  NA  NA          NA    NA       NA  NA  NA  ...
5 Not at all important NA  NA  NA  Interesting Weird Intense  NA  NA  NA  ...
6 Not at all important NA  NA  NA  Good        Weird NA       NA  NA  No  ...
  q86397 q86462 q86699 q363047 CA        gender_orientation race  gender2
1 NA     NA     NA     NA      0.7630798 Hetero_female      White Woman  
2 NA     NA     NA     NA             NA Hetero_male        NA    Man    
3 NA     NA     NA     NA      0.6613091 Hetero_female      NA    Woman  
4 NA     NA     NA     NA             NA Hetero_female      White Woman  
5 NA     NA     NA     NA      0.8754240 Bisexual_female    NA    Woman  
6 NA     N

In [9]:
dim(questions)

[1] 2281   10

In [7]:
head(questions)

  X  
1 q41
2 q42
3 q43
4 q44
5 q45
6 q46
  text                                                                                               
1 How important is religion/God in your life?                                                        
2 Is your duty to religion/God the most important thing in your life?                                
3 Do you believe in some kind of a spiritual, cosmic force outside the realm of most modern religion?
4 Some religions are more correct than others.                                                       
5 Do you believe in ghosts?                                                                          
6 Would you prefer good things happened, or interesting things?                                      
  option_1            option_2           option_3          
1 Extremely important Somewhat important Not very important
2 Yes                 No                                   
3 Yes                 No                                   
4 Tr

The users dataframe stores each user's answers to questions, and the questions dataframe contains each question and its answer choices.

## Question 2

In [11]:
dim(questions)

[1] 2281   10

In [12]:
head(questions)

  X  
1 q41
2 q42
3 q43
4 q44
5 q45
6 q46
  text                                                                                               
1 How important is religion/God in your life?                                                        
2 Is your duty to religion/God the most important thing in your life?                                
3 Do you believe in some kind of a spiritual, cosmic force outside the realm of most modern religion?
4 Some religions are more correct than others.                                                       
5 Do you believe in ghosts?                                                                          
6 Would you prefer good things happened, or interesting things?                                      
  option_1            option_2           option_3          
1 Extremely important Somewhat important Not very important
2 Yes                 No                                   
3 Yes                 No                                   
4 Tr

In [14]:
grep("Google", questions$text)

[1] 2172

In [15]:
questions[2172, ]

  X       text                                       option_1                
1 q170849 Do you Google someone before a first date? Yes. Knowledge is power!
  option_2                   option_3 option_4 N     Type Order
1 No. Why spoil the mystery?                   39621 O         
  Keywords               
1 descriptive; technology

In [19]:
questions$text[grepl("Google", questions$text)]

[1] "Do you Google someone before a first date?"

a. The function returns the line number(s) that the keyword is found on.

b. The word "Google" can be found at row 2,172 in the question "Do you Google someone before a first date?".

## Question 3

In [20]:
dim(users)

[1] 68371  2284

In [21]:
head(users)

  q41                  q42 q43 q45 q46         q48   q49      q50 q55 q56 ...
1 NA                   NA  NA  NA  NA          NA    Carefree NA  No  No  ...
2 NA                   NA  NA  NA  NA          NA    NA       NA  NA  NA  ...
3 Extremely important  NA  NA  NA  Good        NA    Carefree NA  No  NA  ...
4 NA                   NA  NA  NA  NA          NA    NA       NA  NA  NA  ...
5 Not at all important NA  NA  NA  Interesting Weird Intense  NA  NA  NA  ...
6 Not at all important NA  NA  NA  Good        Weird NA       NA  NA  No  ...
  q86397 q86462 q86699 q363047 CA        gender_orientation race  gender2
1 NA     NA     NA     NA      0.7630798 Hetero_female      White Woman  
2 NA     NA     NA     NA             NA Hetero_male        NA    Man    
3 NA     NA     NA     NA      0.6613091 Hetero_female      NA    Woman  
4 NA     NA     NA     NA             NA Hetero_female      White Woman  
5 NA     NA     NA     NA      0.8754240 Bisexual_female    NA    Woman  
6 NA     N

In [24]:
which(names(users) == "q170849")

[1] 1193

In [25]:
users[ , 1193]

      q170849                   
1     No. Why spoil the mystery?
2     NA                        
3     No. Why spoil the mystery?
4     NA                        
5     NA                        
6     Yes. Knowledge is power!  
7     Yes. Knowledge is power!  
8     NA                        
9     NA                        
10    NA                        
11    No. Why spoil the mystery?
12    NA                        
13    Yes. Knowledge is power!  
14    NA                        
15    No. Why spoil the mystery?
16    NA                        
17    NA                        
18    NA                        
19    No. Why spoil the mystery?
20    NA                        
21    NA                        
22    No. Why spoil the mystery?
23    Yes. Knowledge is power!  
24    Yes. Knowledge is power!  
25    NA                        
26    No. Why spoil the mystery?
27    NA                        
28    NA                        
29    NA                        
30    No. 

In [27]:
prop.table(table(users[[1193]], useNA="always"))


No. Why spoil the mystery?   Yes. Knowledge is power! 
                 0.3774115                  0.2020886 
                      <NA> 
                 0.4204999 

a. This corresponds with the questions variable in the users dataframe, specifically q170849.

b. Approximately 37.7% of users do not google someone before the first date.

## Question 4

In [29]:
head(users[[2282]])

[1] "Woman" "Man"   "Woman" "Woman" "Woman" "Man"  

In [31]:
tapply(users[[1193]], users[[2282]], table)

$Man

No. Why spoil the mystery?   Yes. Knowledge is power! 
                     17294                       7549 

$Woman

No. Why spoil the mystery?   Yes. Knowledge is power! 
                      7987                       5660 


In [36]:
tapply(users[[1193]], users[[2282]], function(x) {prop.table(table(x, useNA="always"))})

$Man
x
No. Why spoil the mystery?   Yes. Knowledge is power! 
                 0.4300385                  0.1877160 
                      <NA> 
                 0.3822454 

$Woman
x
No. Why spoil the mystery?   Yes. Knowledge is power! 
                 0.3077605                  0.2180949 
                      <NA> 
                 0.4741446 


## Question 5

In [40]:
table(gsub("_[A-Va-z]*", "", users$gender_orientation))


Bisexual      Gay   Hetero 
    4743     3392    57747 

Markdown notes and sentences and analysis written here.

## Pledge

By submitting this work I hereby pledge that this is my own, personal work. I've acknowledged in the designated place at the top of this file all sources that I used to complete said work, including but not limited to: online resources, books, and electronic communications. I've noted all collaboration with fellow students and/or TA's. I did not copy or plagiarize another's work.

> As a Boilermaker pursuing academic excellence, I pledge to be honest and true in all that I do. Accountable together – We are Purdue.